src/main/java/seedu/address/logic/commands/EditTypeCommand.java

```diff
@@ -0,0 +1,103 @@
+     * edited with {@code editPersonDescriptor}.
+     */
+    protected static Person createEditedPerson(Person personToEdit, EditPersonDescriptor editPersonDescriptor) {
+        assert personToEdit != null;
+
+        Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
+        Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
+        Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
+        Schedule updatedSchedule = editPersonDescriptor.getSchedule().orElse(personToEdit.getSchedule());
+        Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
+
+        return new Person(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedSchedule, updatedTags);
+    }
+
+    /**
+     * Creates and returns an {@code Event} with the details of {@code eventToEdit}
+     * edited with {@code editEventDescriptor}.
+     */
+    protected static Event createEditedEvent(Event eventToEdit, EditEventDescriptor editEventDescriptor) {
+        assert eventToEdit != null;
+
+        EventDescription updatedEventDescription =
+                editEventDescriptor.getEventDescription().orElse(eventToEdit.getEventDescription());
+        LocalDate updatedDate = editEventDescriptor.getDate().orElse(eventToEdit.getDate());
+        LocalTime updatedTime = editEventDescriptor.getTime().orElse(eventToEdit.getTime());
+        Duration updatedDuration = editEventDescriptor.getDuration().orElse(eventToEdit.getDuration());
+
+        return new Event(updatedEventDescription, updatedDate, updatedTime, updatedDuration);
+    }
+
+    /**
+     * Creates and returns a {@code Schedule} with the details of {@code scheduleToEdit}
+     * edited with {@code editEventDescriptor} at {@code targetEventIndex}.
+     */
+    protected static Schedule createEditedSchedule(
+            Schedule scheduleToEdit, Index targetEventIndex, EditEventDescriptor editEventDescriptor) {
+        assert scheduleToEdit != null;
+
+        List<Event> scheduleEvents = scheduleToEdit.getEvents();
+        ArrayList<Event> updatedEvents = new ArrayList<>();
+        EditScheduleDescriptor editScheduleDescriptor = new EditScheduleDescriptor();
+
+        int counter = 0;
+        for (Event toEditEvent : scheduleEvents) {
+            Event updatedEvent;
+            if (counter == targetEventIndex.getZeroBased()) {
+                updatedEvent = createEditedEvent(toEditEvent, editEventDescriptor);
+            } else {
+                updatedEvent = toEditEvent;
+            }
+            updatedEvents.add(updatedEvent);
+            counter++;
+        }
+
+        editScheduleDescriptor.setEvents(updatedEvents);
+
+        return new Schedule(editScheduleDescriptor.getEvents().orElse(scheduleToEdit.getEvents()));
```

Comment:
This part is very weird. Since we explicitly set the events to editScheduleDescriptor before this already, why should we still have to do orElse?